src/main/java/com/couchbase/lite/internal/core/RefCounted.java

```diff
@@ -27,7 +27,7 @@ public synchronized void retain() {
     }
 
     public synchronized void release() {
-        if (--refCount <= 0) { free(); }
+        // if (--refCount <= 0) { free(); }
```

Comment:
The release() is doing nothing now. We don't need this anymore?

Reply:
I think that is true.  Here is my reasoning:

I think that the intention was that call to retain in the `DocContext` would hold the `c4Document` until the Context was no longer needed.  That just doesn't work, though, because there is also a call to `free` in the `AbstractDocument`'s finalizer.  So, the only thing that this `release` call could *ever* do, is to release the document too early.

I am still worried about escape analysis.  It seems quite likely to me that, because there is no way to refer to the c4Document in the DocContext, it will be GCed practically immediately.

Also, FWIW, it is possible to change an `AbstractDocument`'s `c4Document`.  I am not at all convinced that the DocContext points at the right c4Document, if that happens.

Reply:
This is very hard to follow. Probably we could just remove RefCounted class.

Reply:
This is a VF on a release branch.  Changes should be minimal.

Reply:
... and, btw, it *is* gone in 2.8